zipkin-cassandra-core/src/main/java/org/twitter/zipkin/storage/cassandra/Repository.java

```diff
@@ -70,6 +70,23 @@
     private final PreparedStatement selectTraceDurationTail;
     private final Map<String,String> metadata;
 
+    private final ThreadLocal<Set<String>> writtenNames = new ThreadLocal<Set<String>>() {
+                private long hour = TimeUnit.MILLISECONDS.toHours(System.currentTimeMillis());
```

Comment:
In a heavy write scenario would it be possible for the memory usage to grow to an unwieldy size?  In this case maybe it would be worth using an LRUMap (threadsafe), which evicts based on size and might reduce the need to nuke it periodically.  I also heard that the [Guava cache](https://code.google.com/p/guava-libraries/wiki/CachesExplained) is super cool and full featured.  Another possible scenario would be getting cache misses if the data is in the Set in another thread.